src/types/privatekubectl.ts

```diff
@@ -0,0 +1,121 @@
+import { Kubectl } from "./kubectl";
+import { ExecOptions, ExecOutput, getExecOutput } from "@actions/exec";
+import * as core from "@actions/core";
+import * as os from "os";
+import * as fs from "fs";
+
+
+
+export class PrivateKubectl extends Kubectl{
+
+
+  public isPrivate(): boolean {
+      return true;
+  }
+
+  protected async execute(args: string[], silent: boolean = false) {
+    args.unshift("/k8stools/kubectl");
+    var kubectlCmd = args.join(" ");
+    var addFileFlag = false;
+    var eo = <ExecOptions>({ silent });
+
+    if(this.containsFilenames(kubectlCmd)){
+      core.debug("kubectl command contains filenames: " + kubectlCmd);
+      kubectlCmd = kubectlCmd.replace(/[\/][t][m][p][\/]/g ,"");
```

Comment:
what does this do? remove /tmp/ directory?

Reply:
yes. invoke won't allow multiple files unless given as the current directory, i.e .. Full path with tmp was not working, and needed to created a subdir because invoke attempts to process the entire . directory, instead of just yaml file. Files are eventually copied to this new subdir, where only the yamls are present. Only for private clusters